### PR TITLE
Fix #3331 - parse dbSNP CSQ

### LINKED ISSUE #3331: Variant IDs in MIP11 no longer rsID
Scout assumes the ID column of VCF contain dbSNP rsID. As of MIP11 this appears to be a short variant ID similar to the Scout variant display ID instead (`chr_pos_ref_alt`). This breaks linkout to dbSNP, and looks a little funny:
Screenshot 2022-05-06 at 16 38 54

Presumably the intention is to populate the `INFO.dbSNP` field with the rsID, so we should likely switch parsing the dbSNP id to that as an alternative if the ID column does not match `/rs.*/` (cosmic not considered). From MIP11 VCF:
```
##INFO=<ID=dbSNP,Number=A,Type=String,Description="dbSNP ID of the variant">
```
but this field does not seem populated. Likely VEP needs another switch as well, @jemten?

### PR DESCRIPTION
**Pull request by dnil**
This PR adds a functionality or fixes a bug.

Attempt parsing a MIP file, ensure the situation in #3331 with variant ID-like strings in dbSNP button does not occur. If a recent patch of MIP11, VEP CSQ fields should instead be parsed, and dbSNP ids be visible again.

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [x] tests executed by DN on stage and GitHub Actions

### CODE REVIEW

File: scout/parse/variant/variant.py [hunk 1]
```diff
@@ -282,8 +283,9 @@ def parse_variant(
     cosmic_ids = set()
     for parsed_transcript in parse_transcripts(raw_transcripts):
         parsed_transcripts.append(parsed_transcript)
-        for dbsnp in parsed_transcript.get("dbsnp", []):
-            dbsnp_ids.add(dbsnp)
+        for dbsnpkey in ["dbsnp", "rs_dbSNP150", "rs_dbSNP"]:
+            for dbsnp in parsed_transcript.get(dbsnpkey, []):
+                dbsnp_ids.add(dbsnp)
```
Comment (northwestwitch) [inline scout/parse/variant/variant.py:288]: What about duplicates? What if rs_dbSNP150 is rs12345 and rs_dbSNP is also rs12345?

Comment (dnil) [reply scout/parse/variant/variant.py:288]: Right, I guess that's why its a set, though it is not super likely that someone would have both?

Comment (northwestwitch) [reply scout/parse/variant/variant.py:288]: ah yes, of course, it's using add and not append. Sorry, lack of sleep!

**Review by northwestwitch — APPROVED** [review]
Looks good, but of course let's load a MIP11 case to test this pR and the GATK one before deploying!

**Comment by dnil** [comment]
Yes, there should be a patch version for `justhusky` now! I can check.

File: scout/parse/variant/transcript.py [hunk 2]
```diff
@@ -252,13 +252,15 @@ def get_sift_prediction(entry):
...
-    variant_ids = entry.get("EXISTING_VARIATION")
-
-    if variant_ids:
-        for variant_id in variant_ids.split("&"):
-            if variant_id.startswith("rs"):
-                dbsnp_list.append(variant_id)
+    dbsnp_list = set()
+    dbsnp_keys = ["EXISTING_VARIATION", "RS_DBSNP150", "RS_DBSNP"]
+
+    for key in dbsnp_keys:
```
Comment (northwestwitch) [inline scout/parse/variant/transcript.py:258]: Wow looks more complicated now! Perhaps expand also docstring with an example of a complicated entry?

Comment (dnil) [reply scout/parse/variant/transcript.py:258]: Yes, and probably another parse test for a current VEP; the VEP97 plus VEP OldSkool+ ones we carry are not necessarily current.

**Comment by dnil** [comment]
Ok, it works on stage, but I'll add that test for good measure before we merge. 

Before:
Screenshot 2022-05-10 at 14 14 25

After:
Screenshot 2022-05-10 at 16 40 16

**Pull request merged**